settings.gradle

```diff
@@ -45,3 +45,11 @@ buildCache {
         }
     }
 }
+
+gradle.afterProject { project ->
```

Comment:
Why is this executed in `afterProject`, and why not put it in `build.gradle`?

Reply:
We have multiple modules that produce an archive: root project, configuration maven extension. We need to update archiving in both modules. Maybe there will be more modules in the future that also need to have this changed.

Reply:
Ok, but `afterEvaluate` should cause an error in Gradle 8 and it's not compatible with configuration cache ([nr. 8](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:task_configuration_avoidance_general)). 

`afterProject` suffers from the same pitfalls given that it is [doing afterEvaluate](https://github.com/gradle/gradle/blob/d33c5d822d5269e8489fcf53057cd01fa50a0a43/subprojects/core/src/main/java/org/gradle/invocation/DefaultGradle.java#L305).

Reply:
I share Sorin's concerns. `build.gradle` is a "better" place and I don't see a reason why it has to be enclosed in the deprecated `afterEvaluate` block

Reply:
you should create a convention plugin, to be applied by all projects one by one, if you don't want to duplicate this logic on all projects. You'll just have to apply your convention plugin on all projects

Reply:
Great idea! Added convention plugin: https://github.com/jenkinsci/gradle-plugin/pull/220/commits/96827d1be1a07930f2eb83ac3c93eef2fb859b80

Reply:
Refactored to use inputs normalization: https://github.com/jenkinsci/gradle-plugin/pull/220/commits/39c08b42578feaef5c793a1055e7d288a6faa895